src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -85,110 +84,398 @@ abstract class FunctionBuilder[R](parameterTypeInfo: Array[TypeInfo[_]], returnT
     f.put(null, rhs)
   }
 
-  def newArray[T](size: Code[Int])(implicit tti: TypeInfo[T], atti: TypeInfo[Array[T]]): Code[Array[T]] = {
-    new Code[Array[T]] {
-      def emit(il: InsnList): Unit = {
-        size.emit(il)
-        il.add(tti.newArray())
+  def getArg[T](i: Int)(implicit tti: TypeInfo[T]): LocalRef[T] = {
+    assert(i >= 0)
+    assert(i < layout.length)
+    new LocalRef[T](argIndex(i))
+  }
+
+  def classAsBytes(c: Code[R]): Array[Byte] = {
+    mn.instructions.add(start)
+    val l = new mutable.ArrayBuffer[AbstractInsnNode]()
+    c.emit(l)
+    val dupes = l.groupBy(x => x).map(_._2.toArray).filter(_.length > 1).toArray
+    if (dupes.nonEmpty)
+      println(s"dupes: $dupes")
+    l.foreach(mn.instructions.add _)
```

Comment:
Why did you add to an ArrayBuffer and then add instructions?  Just so you can check for dupes?